app/models/group.rb

```diff
@@ -0,0 +1,7 @@
+class Group < ApplicationRecord
+  validates :slug, uniqueness: true
+  validates :slug, :name, :group_type, presence: true
+
+  belongs_to :parent, class_name: 'Group', foreign_key: :parent_group_id, optional: true
+  has_many :children, class_name: 'Group', foreign_key: :parent_group_id
```

Comment:
Not sure if I am fully understanding your question here. In the `has_many` association there is no need to set up `optional: true` as the collection could be empty (leaf node)
Optional is only needed in the `belongs_to` as by default is not true?